Request/ParamFetcher.php

```diff
@@ -323,4 +329,62 @@ private function checkNotIncompatibleParams(Param $config)
             }
         }
     }
+
+    /**
+     * @param Param[] $params
+     */
+    private function resolveParameters(array $params)
+    {
+        foreach ($params as $param) {
+            $param->requirements = $this->resolve($param->requirements);
+            $param->default = $this->resolve($param->default);
+        }
+    }
+
+    private function resolve($value)
+    {
+        if (is_array($value)) {
+            foreach($value as $key => $val) {
+                $value[$key] = $this->resolve($val);
+            }
+
+            return $value;
+        }
+
+        if (!is_string($value)) {
+            return $value;
+        }
+
+        $container = $this->container;
+
+        $escapedValue = preg_replace_callback('/%%|%([^%\s]++)%/', function ($match) use ($container, $value) {
```

Comment:
The logic is taken from the FrameworkBundle router.
Do you think it could be separated from the router to create a parameter resolver service?